kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/richmessaging/ListAlRichMessage.java

```diff
@@ -0,0 +1,102 @@
+
+        if(actionDivider != null) {
+            actionDivider.setVisibility(View.VISIBLE);
+        }
+    }
+
+    @Override
+    protected void setupAlRichMessage(ViewGroup listItemLayout, ALRichMessageModel model) {
+        super.setupAlRichMessage(listItemLayout, model);
+        if (model != null) {
+            if (model.getPayload() != null) {
+                TextView headerText = listItemLayout.findViewById(R.id.headerText);
+                ImageView headerImage = listItemLayout.findViewById(R.id.headerImage);
+                ALRichMessageModel.ALPayloadModel payload = (ALRichMessageModel.ALPayloadModel) GsonUtils.getObjectFromJson(model.getPayload(), ALRichMessageModel.ALPayloadModel.class);
+                if (payload != null) {
+                    RecyclerView listRecycler = listItemLayout.findViewById(R.id.alListItemRecycler);
+                    LinearLayoutManager layoutManager = new LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false);
+                    listRecycler.setLayoutManager(layoutManager);
+                    AlListRMAdapter adapter = (AlListRMAdapter) AlRichMessageAdapterFactory.getInstance().getListRMAdapter(context, message, payload.getElements(), payload.getReplyMetadata(), listener);
+                    listRecycler.setAdapter(adapter);
+
+                    if (!TextUtils.isEmpty(payload.getHeaderText())) {
+                        headerText.setVisibility(View.VISIBLE);
+                        headerText.setText(getHtmlText(payload.getHeaderText()));
+                    } else {
+                        headerText.setVisibility(View.GONE);
+                    }
+
+                    if (!TextUtils.isEmpty(payload.getHeaderImgSrc())) {
+                        headerImage.setVisibility(View.VISIBLE);
+                        Glide.with(context).load(payload.getHeaderImgSrc()).into(headerImage);
+                    } else {
+                        headerImage.setVisibility(View.GONE);
+                    }
+
+                    if (payload.getButtons() != null) {
+                        final List<ALRichMessageModel.AlButtonModel> action = payload.getButtons();
+
+                        if (action.get(0) != null) {
+                            final TextView actionText1 = listItemLayout.findViewById(R.id.actionButton1);
+                            setActionTextView(actionText1, null, 0, payload, action, model);
+                        }
+
+                        if (action.size() > 1 && action.get(1) != null) {
+                            final TextView actionText2 = listItemLayout.findViewById(R.id.actionButton2);
+                            View actionDivider2 = listItemLayout.findViewById(R.id.actionDivider2);
+                            setActionTextView(actionText2, actionDivider2, 1, payload, action, model);
+                        }
+
+                        if (action.size() > 2 && action.get(2) != null) {
```

Comment:
can this go beyond 3 in future?
if yes then we need to write it in a different way so that we do not have to rewrite same lines of code for action1, action2, action3 and so on

Reply:
keeping them three for now as @reytum said